utils.py

```diff
@@ -889,12 +889,30 @@ def check_auth(user, hutch):
     return False
 
 def check_special(ioc, hutch):
-    lines = open(SPECIAL_FILE % hutch).readlines()
-    lines = [l.strip() for l in lines]
-    for l in lines:
-        if l == ioc:
-            return True
-    return False
+    with open(SPECIAL_FILE % hutch) as fp:
+        lines = fp.readlines()
+        lines = [l.strip() for l in lines]
```

Comment:
Only the readlines needs to be in the with.  This is fine though.

Reply:
Generally when accessing a resource that may return null I prefer to have the whole thing scoped. Might be a less than pythonic practice that I brought over from statically typed languages. I agree it doesnt matter too much